### PR TITLE
NGINX avatar serving

### PR DESCRIPTION
**Pull request by Airiuwu**

### CODE REVIEW

**Comment by 7ez** [comment]
Why is the avatar filetype being forced to .jpg lol

**Comment by rose-tinted-glasses** [comment]
> Why is the avatar filetype being forced to .jpg lol

`default_type "text/jpg";`

**Comment by 7ez** [comment]
> > Why is the avatar filetype being forced to .jpg lol
> 
> `default_type "text/jpg";`
> 
> `default_type`

That's not what I meant xd
I was talking about
```nginx
        try_files $uri $uri.jpg /$default_avatar =404;
```

**Comment by Airiuwu** [comment]
when it was running with guweb avatars were jpg

**Comment by 7ez** [comment]
the thing is guweb accepts png jpeg jpg and gif

**Comment by Airiuwu** [comment]
then you make it any different and see if you can

**Comment by minisbett** [comment]
`try_files $uri.png $uri.jpg $uri.jpeg $uri.gif /$default_avatar =404;`

removed $uri because I'm pretty sure there can't be no file extension

File: ext/nginx.conf [hunk 1]
```diff
@@ -10,44 +10,43 @@ upstream bancho {
...
 	# XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
-	server_name assets.cmyui.xyz;
+	server_name assets.your.domain;
 
-	# XXX: you'll need to edit this to match your certificate and key
-	ssl_certificate     /etc/letsencrypt/live/cmyui.xyz/fullchain.pem;
-	ssl_certificate_key /etc/letsencrypt/live/cmyui.xyz/privkey.pem;
-
-	# TODO: further ssl configuration
-	ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
```
Comment (alowave223) [inline ext/nginx.conf:46]: Why? Chipers are needed for Ubuntu 20.04+

File: ext/nginx.conf [hunk 2]
```diff
@@ -10,44 +10,43 @@ upstream bancho {
...
+server {
+    listen 80;
+    server_name a.your.domain;
+
+    location / {
+        root /path/to/bancho.py/.data/avatars;
+
+	set $default_avatar default.png;
+        default_type "text/png";
+        try_files $uri $uri.png /$default_avatar =404;
```
Comment (alowave223) [inline ext/nginx.conf:50]: Is this even tested to work with gif/jpeg/jpg/png files?

File: ext/nginx.conf [hunk 3]
```diff
@@ -10,44 +10,43 @@ upstream bancho {
...
 server {
-	listen 443 ssl;
-
+	listen 80;
 	# XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
-	server_name assets.cmyui.xyz;
+	server_name assets.your.domain;
 
-	# XXX: you'll need to edit this to match your certificate and key
-	ssl_certificate     /etc/letsencrypt/live/cmyui.xyz/fullchain.pem;
```
Comment (alowave223) [inline ext/nginx.conf:42]: Why would you refactor this? It's great example of how path should look and help enduser spent less time in google.

**Review by alowave223 — COMMENTED** [review]
In my mind this PR looks unnecessary and a bit weird because of obscure changes like removing `ssl_chipers`, changing `server_name`. In first place you need to explain what you want to do refator entire config or just add NGINX avatar serving?

File: ext/nginx.conf (hunk 1 above)
Comment (cmyui) [reply ext/nginx.conf:46]: oo in the refactor branch all ssl has been removed - assuming people are using cloudflare flex by default

File: ext/nginx.conf (hunk 2 above)
Comment (cmyui) [reply ext/nginx.conf:50]: yeah we'll need support for other image types

**Comment by Airiuwu** [comment]
current condition all file types that are accepted works any questions???

**Comment by depreciate** [comment]
bruh mkfer stole half the shit i did

**Comment by 7ez** [comment]
Cope

**Review by cmyui — APPROVED** [review]
+1 👍🏻

**Pull request merged**